src/platforms/windows/windowspingsender.cpp

```diff
@@ -24,14 +24,35 @@
 #pragma comment(lib, "Ws2_32")
 
 constexpr WORD WindowsPingPayloadSize = sizeof(quint16);
+constexpr size_t ICMP_ERR_SIZE = 8;
+constexpr size_t MinimumReplyBufferSize = sizeof(ICMP_ECHO_REPLY) +
+                                   WindowsPingPayloadSize + ICMP_ERR_SIZE +
+                                   sizeof(IO_STATUS_BLOCK);
+
+
+#pragma pack(push, 1)
+struct ICMP_ECHO_REPLY_BUFFER {
+  ICMP_ECHO_REPLY reply;
```

Comment:
I filed a comment about it in the JIRA issue, but I think we should also switch out the `ICMP_ECHO_REPLY` struct with an `ICMP_ECHO_REPLY32` as per the [IcmpParseReplies](https://learn.microsoft.com/en-us/windows/win32/api/icmpapi/nf-icmpapi-icmpparsereplies#parameters) documentation. It seems there is some 32/64-bit backwards compatibility issue around the `Data` pointer.

My guess as to the cause of the crash is that for whatever reason the Windows kernel is writing a 32-bit pointer into memory, but because we interpret the struct as a `ICMP_ECHO_REPLY` the client interprets the pointer as being 64-bits, and it just so happens to work fine when the top bits are zero, but crashes otherwise.

Reply:
Or better yet, check if the compiler macros are targeting a 64-bit system and make this an `ICMP_ECHO_REPLY` or `ICMP_ECHO_REPLY32` as appropriate.